Added amqp output

ekini opened the pull request:
Please, review my amqp output plugin.

sparrc commented on outputs/amqp/amqp_test.go:
```diff
@@ -0,0 +1,28 @@
+		t.Skip("Skipping integration test in short mode")
+	}
+
+	var url = "amqp://" + testutil.GetLocalHost() + ":5672/"
```
It looks like you are following the form of the kafka output plugin here. In the case of kafka, we've added a kafka broker to the `scripts/docker-compose.yml` file that the test is connecting to in long mode. I'm not very familiar with AMQP, but is there a similar machine that you could add to the dockerfile? Would a rabbitmq docker container work? I think something like this should work:
```
rabbitmq:
    image: rabbitmq:3-management
    hostname: docker_rabbit
    ports:
        - "15672:15672"
        - "5672:5672"
```

sparrc commented on outputs/amqp/amqp.go:
```diff
@@ -0,0 +1,112 @@
+	// AMQP exchange
+	Exchange string
+	// Routing key
+	RoutingKey string
```
should `routingkey` also be in the sampleConfig?

sparrc commented on outputs/amqp/amqp.go:
```diff
@@ -0,0 +1,112 @@
+
+	for _, p := range bp.Points {
+		// Combine tags from Point and BatchPoints and grab the resulting
+		// line-protocol output string to write to Kafka
```
s/Kafka/amqp/g :-)

sparrc commented:
Could you write a short README for this? See #177

sparrc commented:
This looks great! Thank you very much! Please just address the comments that I've left and I can merge it :+1:

ekini commented:
Thanks for feedback, Cameron! I've corrected typos and added README, but I have one conflict now.
I can do a rebase, but I'm not sure what will happen on github after that.

sparrc commented:
@ekini thanks much for your contribution! I made some tweaks to the documentation and changed the default RoutingTag to "host" instead of "dc", because I thought it would be better to use a tag that is defined by default.

I liked that idea so I also added that same functionality to the Kafka output, since that was just hardcoded to use "host" before.

pdyba commented:
Is it possible to use SSL/Certificates with rabbit.output ?
I did not find appropriate information in docs and in code it self (I am a python developer so I could be wrong :) ) ? 
Is there a chance that it will be added in next version ?

ekini commented:
Yes, I'm going to add it in a week.

ekini commented:
Please, see #536
